src/core/integrate.cpp

```diff
@@ -167,6 +173,20 @@ void integrate_ensemble_init() {
 }
 
 /************************************************************/
+/** @brief Step 1 and 2 of Velocity Verlet scheme
+         v(t+0.5*dt) = v(t) + 0.5*dt * a(t)
+         p(t + dt)   = p(t) + dt * v(t+0.5*dt)
+         NOTE: Depending on the integration method Step 1 and Step 2
+         cannot be combined for the translation.
+*/
+void velocity_verlet_step_1_and_2(ParticleRange &particles) {
+  if (integ_switch == INTEG_METHOD_NPT_ISO) {
```

Comment:
please keep the `integ_switch` stuff in the calling context. This does not belong here, the function does not need to know how it is used.